src/ApparentHorizons/SpherePackIterator.hpp

```diff
@@ -0,0 +1,110 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+///
+/// \file
+/// Defines class SpherePackIterator.
+
+//
+// iterate through an array with the layout of the Ylm-spectral
+// coefficients of SpherePack.
+//
+// Note: SpherePackIterator.cpp contains an explanation of the
+// SpherePack memory layout.
+//================================================================
+#pragma once
+
+#include <ostream>
+#include <vector>
+
+/// Iterates through the coefficients of a SpherePack spectral transform.
+/// This ordering is not intuitive, so SpherePackIterator exists
+/// for the purpose of iterating over the array and determining which
+/// specific values of l and m correspond to each iteration.
+///
+/// SpherePackIterator iterates over each coefficient ONCE, in some
+/// arbitrary order. There are member functions to determine the values
+/// of l and m at each iteration.  Also, operator() returns the position
+/// in the array at each iteration; this position can be used to index
+/// DataMeshes that correspond to spectral coefficients.
+/// The constructor takes the number of collocation points in theta and phi
+/// (that is, in physical space).
+///
+/// Example usage:
+///
+/// \code
+/// // This example sets to zero all coefficients with m == l-1
+/// DataVector data;
+/// for(SpherePackIter it(nth,nph);it;++it) {
+///   if(it->m()==it->l()-1) data[it()] = 0.0;
+/// }
+/// \endcode
+class SpherePackIterator {
+ public:
+  enum CoefficientArray { a, b };
+  SpherePackIterator(const size_t lmax, const size_t mmax,
```

Comment:
`l_max`, `m_max` (unless that conflicts with the member functions)

Reply:
Yes, it does conflict with the member functions.